### PR TITLE
Show correct count for Support entities

### LINKED ISSUE #1777: my-lists: "NaN Support"
If a "Support" entity is saved, the my-lists seem to be broken: 
```
1 Donor 0 Sample 0 Dataset NaN Support
```

### PR DESCRIPTION
**Pull request by mccalluc**
Fix #1777 

Screen Shot 2021-04-27 at 3 02 13 PM

### CODE REVIEW

File: context/app/static/js/components/savedLists/SavedListPanel/SavedListPanel.jsx [hunk 1]
```diff
@@ -2,7 +2,13 @@ import React from 'react';
...
 
 function getEntityCounts(listSavedEntities) {
-  const counts = { Donor: 0, Sample: 0, Dataset: 0 };
+  const counts = new Proxy(
+    // If a "Support" entity appears, we want to count it, but it shouldn't be included otherwise.
+    { Donor: 0, Sample: 0, Dataset: 0 },
+    {
+      get: (target, name) => (name in target ? target[name] : 0),
+    },
+  );
```
Comment (john-conroy) [inline context/app/static/js/components/savedLists/SavedListPanel/SavedListPanel.jsx:11]:
```suggestion
  const counts = {};
```

File: context/app/static/js/components/savedLists/SavedListPanel/SavedListPanel.jsx [hunk 2]
```diff
@@ -2,7 +2,13 @@ import React from 'react';
...
   Object.values(listSavedEntities).forEach((entity) => {
     counts[entity.entity_type] += 1;
```
Comment (john-conroy) [inline context/app/static/js/components/savedLists/SavedListPanel/SavedListPanel.jsx:13]:
```suggestion
    const { entity_type } = entity;
    counts[entity_type] ? counts[entity_type] += 1 : counts[entity_type] = 1;
```

**Review by john-conroy — COMMENTED** [review]
Maybe a bit more immediately understandable, but your call.

File: context/app/static/js/components/savedLists/SavedListPanel/SavedListPanel.jsx (hunk 1 above)
Comment (mccalluc) [reply context/app/static/js/components/savedLists/SavedListPanel/SavedListPanel.jsx:11]: Right now, we always get counts for Donor, Sample, and Dataset, even if they are zero. I think with this suggestion, we'd only see non-zero counts... Does that seem ok to you?

Comment (john-conroy) [reply context/app/static/js/components/savedLists/SavedListPanel/SavedListPanel.jsx:11]: Whoops, the initial counts should be `counts = {Donor: 0, Sample: 0, Dataset: 0};` then.

Comment (john-conroy) [inline context/app/static/js/components/savedLists/SavedListPanel/SavedListPanel.jsx:11]:
```suggestion
  const counts = { Donor: 0, Sample: 0, Dataset: 0 };
```

File: context/app/static/js/components/savedLists/SavedListPanel/SavedListPanel.jsx [hunk 3]
```diff
@@ -4,7 +4,8 @@ import { Panel } from 'js/shared-styles/panels';
 function getEntityCounts(listSavedEntities) {
   const counts = { Donor: 0, Sample: 0, Dataset: 0 };
   Object.values(listSavedEntities).forEach((entity) => {
-    counts[entity.entity_type] += 1;
+    const { entity_type } = entity;
+    counts[entity_type] ? counts[entity_type] += 1 : counts[entity_type] = 1;
```
Comment (mccalluc) [inline context/app/static/js/components/savedLists/SavedListPanel/SavedListPanel.jsx:8]: @john-conroy : This has eslint errors: assignment inside expression can be confusing. I'll use essentially the same logic, but with a real `if`.

**Pull request merged**